pydynamo/storage/disk/sstableiterator.py

```diff
@@ -0,0 +1,165 @@
+from ..error import StorageException
+from ..error import ErrorType
+from ..iterator import Iterator
+from struct import unpack
+
+
+class SStableIterator(Iterator):
+    def __init__(self, sstable_file_name,
+                 index_file_name, index_table, size) -> None:
+        self.cur = None
+        self.sstable_file_name = sstable_file_name
+        self.index_file_name = index_file_name
+        self.index_table = index_table
+        self.cur_offset = -1
+        self.file_size = size
+        self.int_size = 4
+        self.time_size = 26
+
+    def seek(self, key: str) -> None:
+        index_range = self._get_range(key)
+        if len(index_range) == 0:
+            raise StorageException(ErrorType.NOT_FOUND,
+                                   "The key is not found in table.")
+        elif len(index_range) == 1:
+            offset = unpack("i", index_range[0])[0]
+            self.cur_offset = offset
+        elif len(index_range) == 2:
+            offset = self._get_offset(key, index_range[0], index_range[1])
+            if offset is None:
+                raise StorageException(ErrorType.NOT_FOUND,
+                                       "The key is not found in table.")
+            self.cur_offset = offset
+
+    def seek_to_first(self) -> None:
+        self.cur_offset = -1
+
+    def valid(self) -> bool:
+        if self.cur_offset == -1:
+            return True
+        elif self.cur_offset < -1:
+            return False
+        else:
+            file = open(self.sstable_file_name, "rb")
```

Comment:
It doesn't make a lot of sense to open the file each function call: it will create lots of file descriptor; instead, consider only open one file descriptor per Iterator and use `__de__` method to close it on destruction.